src/test/java/org/assertj/core/api/ComparatorFactory_doubleComparatorWithPrecision.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.Comparator;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldBeEqualWithinOffset.shouldBeEqual;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+
+public class ComparatorFactory_doubleComparatorWithPrecision {
+
+  private final ComparatorFactory INSTANCE = ComparatorFactory.INSTANCE;
+
+  @ParameterizedTest
+  @CsvSource({"0.111, 0.110, 0.001, 0.000001",
+              "0.12345, 0.12346, 0.00001, 1e-10",
+              "0.7654321, 0.7654320, 0.0000001, 1e-9",
+              "1.2463, 1.2464, 0.0001, 1e-8"})
+  void should_pass_for_expected_equal_to_actual_in_certain_range(Double expected, Double actual, Double precision, Double smallDifference) {
+    Comparator<Double> comparator0_1d = INSTANCE.doubleComparatorWithPrecision(precision);
+    Double max = Math.max(expected, actual), min = Math.min(expected, actual);
+
+    assertThat(comparator0_1d.compare(max, min)).isEqualTo(0);
+    assertThat(comparator0_1d.compare(min, max)).isEqualTo(0);
+    assertThat(comparator0_1d.compare(max + smallDifference, min)).isEqualTo(1);
+    assertThat(comparator0_1d.compare(min - smallDifference, max)).isEqualTo(-1);
```

Comment:
as commented previously in https://github.com/assertj/assertj-core/pull/2535#discussion_r847202709 this should have only one assertion. I basically wrote the code for that tests, you only have to populate the CsvSource.